src/mongocxx/hint.hpp

```diff
@@ -100,6 +100,8 @@ class MONGOCXX_API hint {
    private:
     stdx::optional<bsoncxx::document::view_or_value> _index_doc;
     stdx::optional<bsoncxx::string::view_or_value> _index_string;
+
+    bsoncxx::document::value create_document() const;
```

Comment:
A better approach would be to create a new public inline function called `to_document_no_deprecated` that is the same as `to_document` but doesn't have the deprecated tag. Then have `to_document` invoke `to_document_no_deprevated`. It also gives users a way out of the same bind. Then when you actually remove it, you take out both.

Also, methods before members, always, and private functions begin with a leading `_`.

Reply:
To the first paragraph:
- That seems like a similar idea to the changes in this PR, except that your ``to_document_no_deprecated`` is called ``create_document``, and that it is a private method instead of a "public inline function". Can you explain why your suggestion is better?

To the second paragraph:
- Is there any *technical* reason why you would want methods before members? Or is it just a convention?
- I thought that in the past you said that you never wanted a leading ``_`` for anything‽

Reply:
- The `_no_deprecated` convention was the planned convention for doing this sort of forwarding function. It clearly expresses the intention, and keeps the word "deprecated" at the call site, making it clearer that you are calling a deprecated function. You could make the decoration `no_deprecated_warning` if you wanted. I'm willing to compromise on that, but the adornment for the "does not issue a deprecation warning" shim should definitely contain the word deprecated so that it can be grepped for.

- There is no technical reason, it is a convention, and one that we have followed everywhere else in the codebase. Please continue to do so.

- The rules are tricky, but a leading underscore followed by [a-z] is OK at class scope, it is only forbidden at global scope. Please see http://en.cppreference.com/w/cpp/language/identifiers. Leading underscore for private members and private functions is the convention in the codebase, unless it would lead to using a reserved identifier.